lib/media/ver10/set_osd.ex

```diff
@@ -0,0 +1,143 @@
+defmodule Onvif.Media.Ver10.SetOSD do
+  import SweetXml
+  import XmlBuilder
+
+  alias Onvif.Media.Ver10.OSD.TextString.FontColor
+  alias Onvif.Media.Ver10.OSD.TextString.BackgroundColor
+  alias Onvif.Media.Ver10.OSD.Image
+  alias Onvif.Device
+  alias Onvif.Media.Ver10.OSD
+
+  @spec soap_action :: String.t()
+  def soap_action, do: "http://www.onvif.org/ver10/media/wsdl/SetOSD"
+
+  @spec request(Device.t(), list) :: {:ok, any} | {:error, map()}
+  def request(device, args),
+    do: Onvif.Media.Ver10.Media.request(device, args, __MODULE__)
+
+  def request_body(%OSD{} = osd) do
+    element(:"s:Body", [
+      element(:"trt:SetOSD", [
+        element(:"trt:OSD", %{token: osd.token}, [
+          element(:"tt:VideoSourceConfigurationToken", osd.video_source_configuration_token),
+          element(
+            :"tt:Type",
+            Keyword.fetch!(Ecto.Enum.mappings(osd.__struct__, :type), osd.type)
+          ),
+          element(:"tt:Position", [
+            element(
+              :"tt:Type",
+              Keyword.fetch!(
+                Ecto.Enum.mappings(osd.position.__struct__, :type),
+                osd.position.type
+              )
+            ),
+            element(:"tt:Pos", %{x: osd.position.pos.x, y: osd.position.pos.y})
+          ]),
+          gen_element_type(osd.type, osd)
+        ])
+      ])
+    ])
+  end
+
+  defp gen_element_type(:text, osd) do
+    element(:"tt:TextString", [
+      element_is_persistent_text(osd.text_string.is_persistent_text),
+      element(
+        :"tt:Type",
+        Keyword.fetch!(
+          Ecto.Enum.mappings(osd.text_string.__struct__, :type),
+          osd.text_string.type
+        )
+      ),
+      element_date_format(osd.text_string.date_format),
+      element_time_format(osd.text_string.time_format),
```

Comment:
These should be removed or the function should return `[]` if `osd.text_string.type` is `plain_text`. We should check and do the same for other types in `osd.text_string.type`

Reply:
@waranlogesh I did not remove it to keep the original data, even if we are not using it. That's why I just pattern matched nil.

Reply:
According to the [spec](https://www.onvif.org/ver10/media/wsdl/media.wsdl#op.SetOSD) it should not be present for plain text type. Even though this implementation passes in Dahua, it might to set the values with explicit error or implicitly in other manufacturers.
 
![image](https://github.com/user-attachments/assets/d8ed9862-888b-43d3-8981-968551241949)